### DISCUSSION TITLE
Topological changes : renumbering for tetra removal

### DISCUSSION THREAD

**Discussion by kleinanita** [sofa-framework/sofa#3156] [Develop in C++] [answered]
Hi guys

A short question maybe for @epernod : we are removing tetra using the topological changes  API of SOFA.
We were wondering how the topological tetrahedron vector is reorganized / renumbered after a tetra is removed
Any idea / guidance ?

SOFA version : master

thanks in advance

**Comment by epernod** [comment, marked as answer]
Hi,

thank you for your question, it makes me realized this is not described anywhere.. I need to update the doc.
So to answer your question. Let say you have a 10 tetra in your topology:
Tetra ids: [0; 1; ...; 9]

Removing tetra id 4 and 6 in the same call/event will be done like this:
1. Warn all components linked to this topology that Tetra [4, 6] are going to be removed
2. Removing tetra 6:
       - swap last tetra (9) and 6
       - pop_back last tetra (9)
3. Same with tetra 4:
       - swap last tetra (8) and 4
       - pop_back last tetra (8)

So at the end you will have a vector of 8 tetra which would be (using initial ids):
[0, 1, 2, 3, 8, 5, 9, 7] 

Let me know if it is clear enough.

**Reply by kleinanita** [reply]
Thank you for your reply
And yes, the explanation is very clear to me !!
